statefun-tasks/src/main/java/com/sbbsystems/statefun/tasks/CallbackFunction.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright [2023] [Frans King, Luke Ashworth]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.sbbsystems.statefun.tasks;
+
+import com.sbbsystems.statefun.tasks.generated.*;
+import com.sbbsystems.statefun.tasks.types.InvalidMessageTypeException;
+import com.sbbsystems.statefun.tasks.types.MessageTypes;
+import org.apache.flink.statefun.sdk.Context;
+import org.apache.flink.statefun.sdk.FunctionType;
+import org.apache.flink.statefun.sdk.StatefulFunction;
+import org.apache.flink.statefun.sdk.annotations.Persisted;
+import org.apache.flink.statefun.sdk.state.PersistedAppendingBuffer;
+import org.apache.flink.statefun.sdk.state.PersistedValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.MessageFormat;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class CallbackFunction implements StatefulFunction {
+    private static final Logger LOG = LoggerFactory.getLogger(CallbackFunction.class);
+    @Persisted
+    private final PersistedAppendingBuffer<TaskResultOrException> batch = PersistedAppendingBuffer.of("batch", TaskResultOrException.class);
+    @Persisted
+    private final PersistedValue<Boolean> pipelineRequestInProgress = PersistedValue.of("pipelineRequestInProgress", Boolean.class);
+    private final FunctionType pipelineFunctionType;
+
+    public CallbackFunction(FunctionType pipelineFunctionType) {
+        this.pipelineFunctionType = pipelineFunctionType;
+    }
+
+    @Override
+    public void invoke(Context context, Object input) {
+        try {
+            if (MessageTypes.isType(input, CallbackSignal.class)) {
+                // signal from PipelineFunction
```

Comment:
Rather than if/else/if/else should we go with with message handler pattern?

Reply:
I went with if/else because the state fields are members of this class and it seemed simpler to manage the interactions from within this class. The logic here is much simpler than the pipeline handling, although it may get a little more complex as we add features.

But let me try refactoring it and moving the persisted fields to another class. I can do the same for the PipelineFunction.